contracts/0.4.24/oracle/LidoOracle.sol

```diff
@@ -73,153 +76,211 @@ contract LidoOracle is ILidoOracle, AragonApp {
+    }
+
+    function setAllowedBeaconBalanceRelativeDecrease(uint256 value) public auth(SET_REPORT_BOUNDARIES) {
+        ALLOWED_BEACON_BALANCE_RELATIVE_DECREASE_POSITION.setStorageUint256(value);
+        emit AllowedBeaconBalanceRelativeDecreaseSet(value);
+    }
 
-        initialized();
+    function getCurrentOraclesReportStatus() public view returns(uint256) {
+        return REPORTS_BITMASK_POSITION.getStorageUint256();
     }
 
+    function getCurrentReportKindsSize() public view returns(uint256) {
+        return gatheredReportKinds.length;
+    }
+
+    function getCurrentReportKind(uint256 index)
+        public view
+        returns(
+            uint128 beaconBalance,
+            uint128 beaconValidators,
+            uint256 count
+        ) {
+        ReportKind storage kind = gatheredReportKinds[index];
+        return (kind.report.beaconBalance, kind.report.beaconValidators, kind.count);
+    }
+
+    /// @dev Initialize function removed from v2 because it is invoked only once
+    function initialize(address, uint64, uint64, uint64, uint64) public {}
+
     /**
-      * @notice Add `_member` to the oracle member committee
-      * @param _member Address of a member to add
-      */
+     * @notice Add `_member` to the oracle member committee
+     * @param _member Address of a member to add
+     */
     function addOracleMember(address _member) external auth(MANAGE_MEMBERS) {
         require(members.length < MAX_MEMBERS, "TOO_MANY_MEMBERS");
         require(address(0) != _member, "BAD_ARGUMENT");
         require(MEMBER_NOT_FOUND == _getMemberId(_member), "MEMBER_EXISTS");
 
         members.push(_member);
-
-        // set quorum to 1 when first member added
-        if (1 == members.length) {
-            QUORUM_POSITION.setStorageUint256(1);
-        }
-
         emit MemberAdded(_member);
-        _assertInvariants();
     }
 
     /**
      * @notice Remove `_member` from the oracle member committee
      * @param _member Address of a member to remove
      */
     function removeOracleMember(address _member) external auth(MANAGE_MEMBERS) {
-        require(members.length > getQuorum(), "QUORUM_WONT_BE_MADE");
-
         uint256 index = _getMemberId(_member);
         require(index != MEMBER_NOT_FOUND, "MEMBER_NOT_FOUND");
-
-        uint256 lastReportedEpochId = (
-            LAST_REPORTED_EPOCH_ID_POSITION.getStorageUint256()
-        );
-
-        MIN_REPORTABLE_EPOCH_ID_POSITION.setStorageUint256(lastReportedEpochId);
         uint256 last = members.length.sub(1);
-
-        uint256 bitMask = gatheredEpochData[lastReportedEpochId].reportsBitMask;
-        if (index != last) {
-            members[index] = members[last];
-            bitMask = bitMask.setBit(index, bitMask.getBit(last));
-        }
-        bitMask = bitMask.setBit(last, false);
-        gatheredEpochData[lastReportedEpochId].reportsBitMask = bitMask;
-
+        if (index != last) members[index] = members[last];
         members.length--;
```

Comment:
And not using SafeMath here? I suggest to stay consistent here.

Reply:
I don't think we need to use safemath at all (if length is 0, `require(MEMBER_NOT_FOUND == _getMemberId(_member), "MEMBER_EXISTS");` will fail) but I must note that using safemath for the first check that substraction is possible and then using regular math where we know it's safe is a consistent pattern.

Reply:
Good point, I agree, but this actually will lead us to remove `SafeMath` at all.